addons/website_sale/models/sale_order.py

```diff
@@ -34,6 +34,15 @@ def _get_note_url(self):
             return website.get_base_url()
         return super()._get_note_url()
 
+    @api.depends('pricelist_id', 'company_id', 'website_id')
+    def _compute_currency_id(self):
+        for order in self:
+            website_currency_id = self.env['website'].get_current_website().currency_id.id
+            if website_currency_id:
+                order.currency_id = order.pricelist_id.currency_id.id or website_currency_id
+            else:
+                super()._compute_currency_id()
+
```

Comment:
This should be removed because the order should be created by the website with the correct pricelist (and updated when needed).

Furthermore, know that get_current_website and website.pricelist_id depends on the session, which shouldn't ever be considered in a stored compute :P

(Also, fyi, the order website is order.website_id, no need to use anything else)